game/engine.py

```
@@ -576,8 +576,19 @@ def get_address():
 
         try:
             conn.send(command)
-            resp = conn.recv()
-            return resp
+            timeout_secs = getattr(
+                self, "_analysis_timeout", self.ANALYSIS_TIMEOUT_SECONDS
+            )
+            if conn.poll(timeout_secs):
+                resp = conn.recv()
+                return resp
```

Ruff (0.15.21)

[warning] Unnecessary assignment to `resp` before `return` statement

Remove unnecessary assignment

(RET504)